src/layouts/Catalogue/Catalogue.js

```diff
@@ -1,358 +1,149 @@
-import React, {Component } from "react";
-import Product from "../../components/Product";
+import React, { useEffect, useReducer, useState } from "react";
+import { Product, NavigationBar, Cart, AddProduct } from "../../components";
 import { ProductApi } from "../../api";
-import { Modal, Button, InputNumber, } from "antd";
-import 'antd/dist/antd.css';
+import "antd/dist/antd.css";
 import styled from "styled-components";
-
-const loading = (
-  <div className="suspense-page">
-    {" "}
-    Loading...{" "}
-  </div>
-);
-
-export default class Catalogue extends Component {
-  state = {
-    menuOpened: false,
-    products: [],
-    selectedProduct: {},
-    onLoading: false,
-    showModal: false,
-    showModalCart: false,
-    cart: [],
-    qty: 1,
-  };
-
-  handleMenuOpen(e) {
-    this.setState({
-      menuOpened: !this.state.menuOpened,
-    });
+import useProductReducer from "../../components/CatalogueReducer";
+import useCartReducer from "../../components/CartReducer";
+
+const Wrapper = styled.div`
+  height: 100%;
+  width: 100%;
+`;
+
+const Main = styled.main`
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+  min-height: calc(100vh - 70px);
+  margin-top: 70px;
+  padding-top: 20px;
+  width: 100%;
+  background: white;
+`;
+
+const Title = styled.h1`
+  text-transform: uppercase;
+  font-size: 50pt;
+  font-weight: 700;
+  text-align: center;
+`;
+
+const Products = styled.ul`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: row;
+  flex-wrap: wrap;
+  width: 75%;
+`;
+
+export default function Catalogue() {
+  const [products, setProducts] = useState([]);
+  const [showModalCart, setShowModalCart] = useState(false);
+
+  const { state, actions } = useProductReducer();
+  const { stateCart, actionCarts } = useCartReducer();
+
+  useEffect(() => {
+    ProductApi.get()
+      .then((res) => res.json())
+      .then((res) => {
+        console.log(`res`, res);
+        setProducts(res);
+      });
+  }, []);
+
+  function showModal(product) {
+    actions.changeValue(product);
   }
```

Comment:
This function has basically become obsolete. Instead of `showModal` you can call `actions.changeValue` and remove this function.